File: back-end/scripts/create_demo_notifications.py
#!/usr/bin/env python3
"""
Script to create demo notifications for user 485458966@qq.com
Run this script to populate the database with sample notifications
"""

import sys
import os
from datetime import datetime, timedelta

# Add the parent directory to the path so we can import app modules
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from app.database import SessionLocal, engine
from app.models import User, Notification
from sqlalchemy.orm import sessionmaker

def create_demo_notifications():
    """Create demo notifications for the specified user"""
    
    # Create database session
    SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
    db = SessionLocal()
    
    try:
        # Find the user by email
        user = db.query(User).filter(User.email == "495458966@qq.com").first()
        
        if not user:
            print("❌ User 485458966@qq.com not found in database")
            return
        
        print(f"✅ Found user: {user.username} (ID: {user.id})")
        
        # Check if user already has notifications
        existing_count = db.query(Notification).filter(Notification.user_id == user.id).count()
        if existing_count > 0:
            print(f"⚠️  User already has {existing_count} notifications")
            response = input("Do you want to add more demo notifications? (y/n): ")
            if response.lower() != 'y':
                print("Cancelled.")
                return
        
        # Demo notifications data
        demo_notifications = [
            {
                "title": "Welcome to ScaleBox! 🎉",
                "message": "Your account has been successfully created. You can now start creating sandboxes and templates.",
                "type": "success",
                "related_entity_type": "account",
                "related_entity_id": str(user.id)
            },
            {
                "title": "New Template Available",
                "message": "A new React TypeScript template has been added to the template library. Check it out!",
                "type": "info",
                "related_entity_type": "template",
                "related_entity_id": "demo-template-1"
            },
            {
                "title": "System Maintenance Notice",
                "message": "Scheduled maintenance will occur on Sunday at 2 AM UTC. Service may be temporarily unavailable.",
                "type": "warning",
                "related_entity_type": "system",
                "related_entity_id": "maintenance-2024-01"
            },
            {
                "title": "API Key Expiring Soon",
                "message": "Your API key 'production-key-1' will expire in 7 days. Please rotate it soon.",
                "type": "warning",
                "related_entity_type": "api_key",
                "related_entity_id": "key-123"
            },
            {
                "title": "Sandbox Created Successfully",
                "message": "Your new sandbox 'my-test-project' has been created and is ready to use.",
                "type": "success",
                "related_entity_type": "sandbox",
                "related_entity_id": "sandbox-456"
            },
            {
                "title": "Billing Update",
                "message": "Your monthly usage is at 85% of your plan limit. Consider upgrading for more resources.",
                "type": "info",
                "related_entity_type": "billing",
                "related_entity_id": "billing-789"
            },
            {
                "title": "Security Alert",
                "message": "New login detected from a new device. If this wasn't you, please review your account security.",
                "type": "error",
                "related_entity_type": "security",
                "related_entity_id": "login-alert-001"
            }
        ]
        
        # Create notifications with different timestamps
        created_count = 0
        for i, notification_data in enumerate(demo_notifications):
            # Create timestamps that are progressively older
            created_at = datetime.utcnow() - timedelta(hours=i*2)
            
            notification = Notification(
                user_id=user.id,
                title=notification_data["title"],
                message=notification_data["message"],
                type=notification_data["type"],
                is_read=False,  # Make them unread to show the badge
                related_entity_type=notification_data["related_entity_type"],
                related_entity_id=notification_data["related_entity_id"],
                created_at=created_at
            )
            
            db.add(notification)
            created_count += 1
            print(f"📝 Created notification: {notification_data['title']}")
        
        # Commit all notifications
        db.commit()
        print(f"✅ Successfully created {created_count} demo notifications for user {user.email}")
        print("🔔 The notification bell should now show a badge with the unread count")
        
    except Exception as e:
        print(f"❌ Error creating demo notifications: {e}")
        db.rollback()
    finally:
        db.close()

if __name__ == "__main__":
    print("🚀 Creating demo notifications for user 485458966@qq.com...")
    create_demo_notifications()
    print("✨ Done!") 

File: front-end/scripts/create-demo-notifications.js
// Script to create demo notifications for user 485458966@qq.com
// Run this in the browser console after logging in as 485458966@qq.com

async function createDemoNotifications() {
  try {
    // Get the auth token from localStorage
    const token = localStorage.getItem('auth-token');
    if (!token) {
      console.error('❌ No auth token found. Please log in as 485458966@qq.com first.');
      return;
    }

    console.log('🔑 Found auth token, creating demo notifications...');

    // Call the demo endpoint
    const response = await fetch('http://43.199.160.148:8000/api/notifications/demo', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
    });

    if (response.ok) {
      const result = await response.json();
      console.log('✅ Demo notifications created successfully!', result);
      console.log('📱 Check the notification bell icon in the header to see the new notifications.');
      console.log('🔔 You should now see a number badge on the notification icon.');
      
      // Refresh the page to show the new notifications
      setTimeout(() => {
        console.log('🔄 Refreshing page to show new notifications...');
        window.location.reload();
      }, 2000);
    } else {
      const errorData = await response.json();
      console.error('❌ Failed to create demo notifications:', errorData);
    }
  } catch (error) {
    console.error('❌ Error creating demo notifications:', error);
  }
}

// Alternative: Create notifications directly via API calls
async function createDirectNotifications() {
  try {
    const token = localStorage.getItem('auth-token');
    if (!token) {
      console.error('❌ No auth token found. Please log in first.');
      return;
    }

    const demoNotifications = [
      {
        title: "Welcome to ScaleBox! 🎉",
        message: "Your account has been successfully created. You can now start creating sandboxes and templates.",
        type: "success"
      },
      {
        title: "New Template Available",
        message: "A new React TypeScript template has been added to the template library. Check it out!",
        type: "info"
      },
      {
        title: "System Maintenance Notice",
        message: "Scheduled maintenance will occur on Sunday at 2 AM UTC. Service may be temporarily unavailable.",
        type: "warning"
      },
      {
        title: "API Key Expiring Soon",
        message: "Your API key 'production-key-1' will expire in 7 days. Please rotate it soon.",
        type: "warning"
      },
      {
        title: "Sandbox Created Successfully",
        message: "Your new sandbox 'my-test-project' has been created and is ready to use.",
        type: "success"
      }
    ];

    console.log('📝 Creating demo notifications...');

    for (const notification of demoNotifications) {
      const response = await fetch('http://43.199.160.148:8000/api/notifications/', {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(notification),
      });

      if (response.ok) {
        console.log(`✅ Created notification: ${notification.title}`);
      } else {
        console.error(`❌ Failed to create notification: ${notification.title}`);
      }
    }

    console.log('🎉 All demo notifications created! Check the notification bell icon.');
    
    // Refresh after a short delay
    setTimeout(() => {
      window.location.reload();
    }, 1000);
  } catch (error) {
    console.error('❌ Error creating notifications:', error);
  }
}

// Export functions for use in console
window.createDemoNotifications = createDemoNotifications;
window.createDirectNotifications = createDirectNotifications;

console.log('📋 Demo notification functions loaded!');
console.log('💡 Run createDemoNotifications() to use the demo endpoint');
console.log('💡 Run createDirectNotifications() to create notifications directly'); 